WarehouseManagement.Core/Services/DeliveryService.cs

```diff
@@ -0,0 +1,299 @@
+                        ? d.IsApproved
+                            ? DeliveryStatus.Approved.ToString()
+                            : DeliveryStatus.Finished.ToString()
+                        : d.Entries.Any(e => e.StartedProccessing.HasValue)
+                            ? DeliveryStatus.Processing.ToString()
+                            : DeliveryStatus.Waiting.ToString()
+                    : DeliveryStatus.Waiting.ToString(),
+                Entries = d
+                    .Entries.Select(e => new DeliveryEntryDto()
+                    {
+                        Id = e.Id,
+                        FinishedProccessing = e.FinishedProccessing,
+                        StartedProccessing = e.StartedProccessing,
+                        ZoneId = e.ZoneId
+                    })
+                    .ToList(),
+                Markers = d
+                    .DeliveriesMarkers.Where(dm => dm.DeliveryId == id)
+                    .Select(dm => new DeliveryMarkerDto()
+                    {
+                        MarkerId = dm.MarkerId,
+                        MarkerName = dm.Marker.Name
+                    })
+                    .ToList()
+            })
+            .FirstOrDefaultAsync();
+
+        if (delivery == null)
+        {
+            throw new KeyNotFoundException(id.ToString());
+        }
+
+        return delivery;
+    }
+
+    public async Task<ICollection<DeliveryDto>> GetAllAsync(PaginationParameters paginationParams)
+    {
+        Expression<Func<Delivery, bool>> filter = v =>
+            EF.Functions.Like(v.ReceptionNumber, $"%{paginationParams.SearchQuery}%")
+            || EF.Functions.Like(v.SystemNumber, $"%{paginationParams.SearchQuery}%")
+            || EF.Functions.Like(v.TruckNumber, $"%{paginationParams.SearchQuery}%");
+
+        var deliveries = await repository
+            .AllReadOnly<Delivery>()
+            .Paginate(paginationParams, filter)
+            .Select(d => new DeliveryDto()
+            {
+                Id = d.Id,
+                Cmr = d.Cmr,
+                DeliveryTime = d.DeliveryTime,
+                IsApproved = d.IsApproved,
+                Packages = d.Packages,
+                Pallets = d.Pallets,
+                Pieces = d.Pieces,
+                ReceptionNumber = d.ReceptionNumber,
+                SystemNumber = d.SystemNumber,
+                TruckNumber = d.TruckNumber,
+                VendorId = d.VendorId,
+                VendorName = d.Vendor.Name,
+                Status = d.Entries.Any()
+                    ? d.Entries.All(e => e.FinishedProccessing.HasValue)
+                        ? d.IsApproved
+                            ? DeliveryStatus.Approved.ToString()
+                            : DeliveryStatus.Finished.ToString()
+                        : d.Entries.Any(e => e.StartedProccessing.HasValue)
+                            ? DeliveryStatus.Processing.ToString()
+                            : DeliveryStatus.Waiting.ToString()
```

Comment:
not sure how C# works, but cant we have this too as some kind of a util or just a seperate function? `GetDeliveryStatus` for example?

Reply:
let's keep it like this right now and if need it we will think how to implement it as a separate function.